index.js

```diff
@@ -2,25 +2,30 @@
 const { ProvidePlugin } = require("webpack")
 const filterObject = require("filter-obj")
 
-const excludeObjectKeys = (object, excludeKeys) => filterObject(object, key => !excludeKeys.includes(key))
+const filterObjectKeys = (object, keys, isInclude) => filterObject(object, key => keys.includes(key) === isInclude)
 
 module.exports = class NodePolyfillPlugin {
-	constructor(options = {}) {
-		this.options = {
-			excludeAliases: [],
-			...options
+	constructor({ excludeAliases = [], includeAliases = [] } = {}) {
+		if (includeAliases.length > 0 && excludeAliases.length > 0) {
+			throw new Error("Unable to use excludeAliases and includeAliases options!")
 		}
+
+		this.options = { excludeAliases, includeAliases }
 	}
 
 	apply(compiler) {
-		compiler.options.plugins.push(new ProvidePlugin(excludeObjectKeys({
+		const isInclude = this.options.includeAliases.length > 0
+		const keys = isInclude ? this.options.includeAliases : this.options.excludeAliases
+		const filterFunc = object => filterObjectKeys(object, keys, isInclude)
```

Comment:
`filterObjectKeys` should be a curried function that is provided both `includeAliases` and `excludeAliases` and returns a `filter` function that does it.

Both the inclusion and exclusion predicates should be created in a ternary expression and passed to `filterObject` within the curried function.

Reply:
The ladder part of this requested change is a bit unclear to me, I landed on the following as being the most terse. 

```js
const filterObjectKeys = (includeAliases, excludeAliases) => {
	const isInclude = includeAliases.length > 0
	const keys = isInclude ? includeAliases : excludeAliases
	return object => filterObject(object, key => keys.includes(key) === isInclude)
}
```

Reply:
The code is too terse for its own good. This is much easier to understand:

```js
if (includeAliases.length > 0) {
	return filterObject(object, key => includeAliases.includes(key))
}

return filterObject(object, key => !excludeAliases.includes(key))
```